src/bluehawk/bluehawk.ts

```diff
@@ -17,7 +17,10 @@ import { isBinary } from "istextorbinary";
 export class Bluehawk {
   // Register the given command on the processor and validator. This enables
   // support for the command under the given name.
-  registerCommand(name: string, command: AnyCommand): void {
+  registerCommand<Command extends AnyCommand>(
+    name: string,
+    command: Command
+  ): void {
```

Comment:
After looking at this and uses of it throughout the code base, I think Commands should be refactored to contain their name. I didn't see any uses of a command where their name wasn't required, except for the creation of the `state-uncomment` command which cherry-picks functionality from other commands and could do so with the name integrated into the object. 

This should improve developer ergonomics in the future because all dependent information is stored within the object.

Reply:
The original reason to split was so that clients can register commands under different names, which is useful for aliasing. I could see it your way though. We could just move the name to the Command and then add a first class "alias" function.

Reply:
Added to trello